3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -27,9 +27,11 @@ exercise file.
 const rollDie = require('../../helpers/pokerDiceRoller');
 
 function rollDice() {
-  // TODO Refactor this function
   const dice = [1, 2, 3, 4, 5];
-  return rollDie(1);
+  const promises = dice.map((diceItem) => {
+    return rollDie(diceItem);
+  });
```

Comment:
The `.map()` method takes a callback function that takes a die (number) as an argument and returns a promise for that die. This function "signature" perfectly matches that of the `rollDie` function. Therefore the code can be simplified to:

```suggestion
  const promises = dice.map(rollDie);
```

Reply:
Thank you Jim. Updated my code as you mentioned.